### ISSUE TITLE
Problems with Dynamic batching

### ISSUE THREAD

**Issue by 2Bye** [triton-inference-server/server#7029]
Hi all! I'm trying to implement dynamic batching for my model, but I can't solve this problem. I read the article, read the documentation, but I don’t understand how to solve this problem.

My ONNX model accepts 4 inputs

```
(torch.Size([1, 37]), torch.Size([1]), torch.Size([1]), torch.Size([1, 1024]))
```

But I can give it such inputs
```
(torch.Size([2, 37]), torch.Size([2]), torch.Size([2]), torch.Size([2, 1024]))
```

And my output seems like
```torch.Size([2, 1, 33536]) or torch.Size([1, 1, 33536]) or torch.Size([X, 1, 33536])```

If I write in the config

```
max_batch_size: 16
dynamic_batching {
    max_queue_delay_microseconds: 100
}
```

I get error:

**_failed to load 'model' version 1: Invalid argument: model 'model', tensor 'x_tst': the model expects 2 dimensions (shape [-1,-1]) but the model configuration specifies 3 dimensions (an initial batch dimension because max_batch_size > 0 followed by the explicit tensor shape, making complete shape [-1,-1,-1])_**

My model has the ability to count batches, but I don't understand how to write a config for this scenario.

Docker start command: 
```sh
docker run --name triton --gpus '1' --rm -it --network host -v "$PWD"/model_repository:/models nvcr.io/nvidia/tritonserver:22.07-py3 tritonserver --model-repository=/models
```

Please tell me how I can enable **Dynamic batching** it for my model
My config

```config
backend: "onnxruntime"
max_batch_size: 16
dynamic_batching {
    max_queue_delay_microseconds: 100
}
input [
    {
        name: "x_tst"
        data_type: TYPE_INT64
        dims: [-1, -1]
    },
    {
        name: "x_tst_lengths"
        data_type: TYPE_INT64
        dims: [-1]
    },
    {
        name: "sid"
        data_type: TYPE_INT64
        dims: [1]
    },
    {
        name: "emo"
        data_type: TYPE_FP32
        dims: [-1, 1024]
    }
]
output [
    {
        name: "output",
        data_type: TYPE_FP32,
        dims: [-1,-1],
    }
]
```

**Comment by rmccorm4** [comment]
Hi @2Bye, when using dynamic batching (specifying a `max_batch_size > 0`), the batch dimension is implicitly included in the input/output configs. So if you have a 2-dimensional input where the 1st dimension is the batch dimension `[BATCH_SIZE, X]`, then it should look like:
```
max_batch_size: 16
...
input [
    {
        name: "x_tst"
        data_type: TYPE_INT64
        dims: [-1]        # batch dimension is implicitly included by Triton when max_batch_size > 0
    }
...
]
```

and if you weren't using dynamic batching, then it would look like:
```
max_batch_size: 0
...
input [
    {
        name: "x_tst"
        data_type: TYPE_INT64
        dims: [-1, -1]          # batch dimension is explicitly included by user when max_batch_size == 0
    }
...
]
```

This applies to all inputs/outputs in the model config.

Hope this helps.

**Comment by pvijayakrish** [comment]
Thanks @rmccorm4. 
@2Bye Here are some references that could help with understanding [max_batch_size](https://github.com/triton-inference-server/server/issues/3527#issuecomment-962213150 ) and [about shape tensors in model config](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#shape-tensors).

**Comment by 2Bye** [comment]
@pvijayakrish @rmccorm4 Thanks for your reply.

Correct me if I wrong, Now I need change input to my models? I need change 2 inputs
```
    {
        name: "x_tst_lengths"
        data_type: TYPE_INT64
        dims: [-1]
    },
    {
        name: "sid"
        data_type: TYPE_INT64
        dims: [1]
    },
```

Now it is like
```(torch.Size([2, 37]), torch.Size([2]), torch.Size([2]), torch.Size([2, 1024]))```

But I need change input like this
```(torch.Size([2, 37]), torch.Size([2, 1]), torch.Size([2, 1]), torch.Size([2, 1024]))```

and then I can use Dynamic and batch inference. Finally my config must be like this?
```
backend: "onnxruntime"
max_batch_size: 16
dynamic_batching {
    max_queue_delay_microseconds: 100
}
input [
    {
        name: "x_tst"
        data_type: TYPE_INT64
        dims: [-1] ### it transforms to [-1, -1]
    },
    {
        name: "x_tst_lengths"
        data_type: TYPE_INT64
        dims: [1] ### it transforms to [-1, 1]
    },
    {
        name: "sid"
        data_type: TYPE_INT64
        dims: [1] ### it transforms to [-1, 1]
    },
    {
        name: "emo"
        data_type: TYPE_FP32
        dims: [1024] ### it transforms to [-1, 1024]
    }
]
output [
    {
        name: "output",
        data_type: TYPE_FP32,
        dims: [1, -11], ### it transforms to [-1, 1, -1]
    }
]
```

**Comment by manalijain18** [comment]
STATUS - UNAVAILABLE: Invalid argument: model 'mistral_7b', tensor 'onnx::Neg_2': for the model to support batching the shape should have at least 1 dimension and the first dimension must be -1; but shape expected by the model is []  

config is - 
name: "mistral_7b"
platform: "onnxruntime_onnx"
max_batch_size: 8  # Set your maximum batch size here (e.g., 8)
input [
  {
    name: "input_ids"
    data_type: TYPE_INT64
    dims: [ -1 ]  # Model expects a batch of sequences (2D tensor)
  },
  {
    name: "attention_mask"
    data_type: TYPE_INT64
    dims: [ -1 ]  # Model expects a batch of sequences (2D tensor)
  },
  {
    name: "onnx::Neg_2"
    data_type: TYPE_INT64
    dims: [ 1 ]  # Scalar input
  }
]
output [
  {
    name: "logits"
    data_type: TYPE_FP32
    dims: [ -1, 32768 ]  # Adjust based on your model's output dimensions
  }
]
how to fix it?

**Comment by manalijain18** [comment]
if i handled Inputs:
Name: input_ids, Shape: [-1, -1]
Name: attention_mask, Shape: [-1, -1]
Name: onnx::Neg_2, Shape: [-1, -1]
Name: onnx::Neg_3, Shape: []       this is onnx::Neg_3, Shape: []    created,   how to handle this?
